src/controllers/midi/midicontroller.cpp

```diff
@@ -160,53 +160,78 @@ void MidiController::destroyOutputHandlers() {
     }
 }
 
-QString formatMidiMessage(const QString& controllerName,
-                          unsigned char status, unsigned char control,
-                          unsigned char value, unsigned char channel,
-                          unsigned char opCode, mixxx::Duration timestamp) {
+QString MidiController::formatMidiMessage(const QString& controllerName,
+                                          unsigned char status, unsigned char control,
+                                          unsigned char value, unsigned char channel,
+                                          unsigned char opCode, mixxx::Duration timestamp) {
+    QString msg2;
+    if (timestamp == mixxx::Duration::fromMillis(0)) {
+      msg2 = "outgoing:";
+    } else {
+      msg2 = QString("t:%1").arg(timestamp.formatMillisWithUnit());
+    }
     switch (opCode) {
         case MIDI_PITCH_BEND:
-            return QString("%1: t:%2 status 0x%3: pitch bend ch %4, value 0x%5")
-                    .arg(controllerName, timestamp.formatMillisWithUnit(),
+            return QString("%1: %2 status 0x%3: pitch bend ch %4, value 0x%5")
+                    .arg(controllerName, msg2,
                          QString::number(status, 16).toUpper(),
                          QString::number(channel+1, 10),
                          QString::number((value << 7) | control, 16).toUpper().rightJustified(4,'0'));
         case MIDI_SONG_POS:
             return QString("%1: t:%5 status 0x%3: song position 0x%4")
```

Comment:
Is the `%5` a typo? There are only 4 arguments. It seems it should be changed to `%2`.